src/Altinn.Platform/Altinn.Platform.Storage/Storage/Controllers/InstancesController.cs

```diff
@@ -262,55 +278,61 @@ public async Task<ActionResult> Get(int instanceOwnerId, Guid instanceGuid)
         /// </summary>
         /// <param name="appId">the applicationid</param>
         /// <param name="instanceOwnerId">instance owner id</param>
-        /// <param name="instanceTemplate">The instance template to base the new instance on</param>
         /// <returns>instance object</returns>
         /// <!-- POST /instances?appId={appId}&instanceOwnerId={instanceOwnerId} -->
         [HttpPost]
-        public async Task<ActionResult> Post(string appId, int? instanceOwnerId, [FromBody] Instance instanceTemplate)
-        {                       
+        [DisableFormValueModelBinding]
+        public async Task<ActionResult> Post(string appId, int? instanceOwnerId)
+        {
             // check if metadata exists
             Application appInfo = GetApplicationOrError(appId, out ActionResult appInfoErrorResult);
             if (appInfoErrorResult != null)
             {
                 return appInfoErrorResult;
             }
 
+            Instance instanceTemplate = await ReadInstanceTemplateFromBody(Request, appInfo);
+            
             // get instanceOwnerId from three possible places
             int ownerId = GetOrLookupInstanceOwnerId(instanceOwnerId, instanceTemplate, out ActionResult instanceOwnerErrorResult);
+
             if (instanceOwnerErrorResult != null)
             {
                 return instanceOwnerErrorResult;
-            }            
+            }
 
             if (instanceTemplate == null)
             {
                 instanceTemplate = new Instance();
-            }           
+            }
 
             DateTime creationTime = DateTime.UtcNow;
-
             string org = appInfo.Org;
             string userName = GetUser(User);
 
-            Instance createdInstance = createdInstance = new Instance()
-                {
-                    InstanceOwnerId = ownerId.ToString(),
-
-                    CreatedBy = userName,
-                    CreatedDateTime = creationTime,
-                    LastChangedBy = userName,
-                    LastChangedDateTime = creationTime,
-                    AppId = appId,
-                    Org = org,
-
-                    VisibleDateTime = DateTimeHelper.ConvertToUniversalTime(instanceTemplate.VisibleDateTime),
-                    DueDateTime = DateTimeHelper.ConvertToUniversalTime(instanceTemplate.DueDateTime),
-                    Labels = instanceTemplate.Labels,
-                    PresentationField = instanceTemplate.PresentationField,
-
-                    InstanceState = new InstanceState { IsArchived = false, IsDeleted = false, IsMarkedForHardDelete = false },
-                };
-           
+            Instance createdInstance = new Instance()
+            {
+                Id = instanceTemplate.Id,
```

Comment:
This is not good! Storage should be the only one responsible for generating the guid.